internal/containerd/install.go

```diff
@@ -127,3 +132,75 @@ func isContainerdNotInstalled() bool {
 	_, runcNotFoundErr := exec.LookPath(runcPackageName)
 	return containerdNotFoundErr != nil || runcNotFoundErr != nil
 }
+
+// Client is a containerd runtime client wrapper
+// Holds the internalapi.RuntimeService for pod/container operations
+type Client struct {
+	Runtime internalapi.RuntimeService
+}
+
+// NewClient creates a new Client with a real containerd runtime service
+func NewClient() (*Client, error) {
+	runtime, err := remote.NewRuntimeService(ContainerRuntimeEndpoint, 5*time.Second)
+	if err != nil {
+		return nil, err
+	}
+	return &Client{Runtime: runtime}, nil
+}
+
+func (c *Client) RemovePods() error {
+	podSandboxes, err := c.Runtime.ListPodSandbox(&v1.PodSandboxFilter{
+		State: &v1.PodSandboxStateValue{
+			State: v1.PodSandboxState_SANDBOX_READY,
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	for _, sandbox := range podSandboxes {
+		zap.L().Info("Stopping pod..", zap.String("pod", sandbox.Metadata.Name))
+		err := util.RetryExponentialBackoff(3, 2*time.Second, func() error {
+			if err := c.Runtime.StopPodSandbox(sandbox.Id); err != nil {
+				return err
+			}
+			if err := c.Runtime.RemovePodSandbox(sandbox.Id); err != nil {
+				return err
+			}
+			return nil
+		})
+		if err != nil {
+			zap.L().Info("ignored error stopping pod", zap.Error(err))
+		}
+	}
+
+	// If pod sandbox deletes dont work, we can try to stop and remove containers individually
+	containers, err := c.Runtime.ListContainers(nil)
```

Comment:
what is the var we are pasing nil for?

Reply:
This is a filter for list containers call. We also want to remove stopped containers here, since post containerd daemon stop, stopped containers are not anymore GCed.